pkg/controller/controller.go

```diff
@@ -145,6 +164,26 @@ func (c *controller) brokerDelete(obj interface{}) {
 	glog.V(4).Infof("Received delete event for Broker %v", broker.Name)
 }
 
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+// It enforces that the reconcileBrokerKey is never invoked concurrently with the same key.
+func (c *controller) brokerWorker() {
+	for {
+		func() {
```

Comment:
We *know* where the exit is, and there is only *one* exit, so just call the `Done` there vs in a `defer`.